### PR TITLE
build: introduce script to update dev-infra browser mirror 

### PR DESCRIPTION
**Pull request by devversion**
Adds a script to ease the browser mirror uploading in the future.

Example:

```
✅ Downloaded: chromium - linux_x64 driver.
✅ Downloaded: chromium - mac_arm64 driver.
✅ Downloaded: chromium - mac_x64 driver.
✅ Downloaded: chromium - windows_x64 driver.
✅ Downloaded: chromium - mac_arm64 browser.
✅ Downloaded: chromium - mac_x64 browser.
✅ Downloaded: chromium - windows_x64 browser.
✅ Downloaded: chromium - linux_x64 browser.

Fetched all browser artifacts. Now uploading to mirror.

✅ Uploaded: mac_x64 driver-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/mac_x64/driver-bin.zip
✅ Uploaded: windows_x64 driver-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/windows_x64/driver-bin.zip
✅ Uploaded: mac_arm64 driver-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/mac_arm64/driver-bin.zip
✅ Uploaded: linux_x64 driver-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/linux_x64/driver-bin.zip
✅ Uploaded: mac_x64 browser-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/mac_x64/browser-bin.zip
✅ Uploaded: windows_x64 browser-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/windows_x64/browser-bin.zip
✅ Uploaded: linux_x64 browser-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/linux_x64/browser-bin.zip
✅ Uploaded: mac_arm64 browser-bin
  -> https://storage.googleapis.com/dev-infra-mirror/chromium/950435/mac_arm64/browser-bin.zip
Uploaded chromium artifacts to the Google Cloud Storage mirror.
```

### CODE REVIEW

File: bazel/browsers/update-script/upload-mirror.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+/** Name of the Google Cloud Storage bucket for the browser mirror. */
+const MIRROR_BUCKET_NAME = 'dev-infra-mirror';
+
+/** Gets the directory in the mirror bucket for a given browser instance. */
+export function getMirrorDirectoryForBrowserInstance<T>(browser: Browser<T>): string {
+  return `${browser.name}/${browser.revision}`;
+}
+
+/** Uploads a browser platform artifact to the browser mirror. */
+export async function uploadArtifactToMirror<T>(
```
Comment (josephperrott) [inline bazel/browsers/update-script/upload-mirror.ts:25]: Does this generic get used?

File: bazel/browsers/update-script/upload-mirror.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          filePath: driverTmpPath,
+          type: 'driver-bin',
+        })),
+    );
+  }
+
+  const tasks = await Promise.all(downloadTasks);
+  const uploadTasks: Promise<{platform: Platform; type: ArchiveType; file: File}>[] = [];
+
+  console.info('');
```
Comment (josephperrott) [inline bazel/browsers/update-script/upload-mirror.ts:93]:
```suggestion
  console.info();
```

**Review by josephperrott — APPROVED** [review]
LGTM

File: bazel/browsers/update-script/upload-mirror.ts (hunk 1 above)
Comment (devversion) [reply bazel/browsers/update-script/upload-mirror.ts:25]: good catch.

**Pull request merged**